src/messaging/tests/MessagingContext.cpp

```diff
@@ -35,10 +35,22 @@ CHIP_ERROR MessagingContext::Init(nlTestSuite * suite, TransportMgrBase * transp
     chip::Transport::AdminPairingInfo * destNodeAdmin = mAdmins.AssignAdminId(mDestAdminId, GetDestinationNodeId());
     VerifyOrReturnError(destNodeAdmin != nullptr, CHIP_ERROR_NO_MEMORY);
 
-    ReturnErrorOnFailure(mSecureSessionMgr.Init(GetSourceNodeId(), &GetSystemLayer(), transport, &mAdmins));
+    if (useDummyMessageCounter)
+    {
+        ReturnErrorOnFailure(
+            mSecureSessionMgr.Init(GetSourceNodeId(), &GetSystemLayer(), transport, &mAdmins, &mDummyMessageCounterManager));
+    }
+    else
+    {
+        ReturnErrorOnFailure(
+            mSecureSessionMgr.Init(GetSourceNodeId(), &GetSystemLayer(), transport, &mAdmins, &mMessageCounterManager));
+    }
 
     ReturnErrorOnFailure(mExchangeManager.Init(&mSecureSessionMgr));
 
+    if (!useDummyMessageCounter)
+        ReturnErrorOnFailure(mMessageCounterManager.Init(&mExchangeManager));
```

Comment:
Dummy should support Init() call as well. Why the condition?

Reply:
done